xchange-okcoin/src/test/java/info/bitrich/xchangestream/okcoin/OkCoinV3StreamingMarketDataServiceTest.java

```diff
@@ -0,0 +1,50 @@
+package info.bitrich.xchangestream.okcoin;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.reactivex.Completable;
+import io.reactivex.Observable;
+import io.reactivex.observers.TestObserver;
+import org.junit.Before;
+import org.junit.Test;
+import org.knowm.xchange.currency.CurrencyPair;
+import org.knowm.xchange.dto.Order;
+import org.knowm.xchange.dto.marketdata.OrderBook;
+import org.knowm.xchange.dto.trade.LimitOrder;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+public class OkCoinV3StreamingMarketDataServiceTest {
```

Comment:
We avoid tests in the normal build talking directly to exchanges, since it means our builds fail if the exchange is down.  Tests like these are usually modelled as runnable "Example" classes (plenty of examples to follow).

One thing we should do more of is full-stack tests using utilities like Wiremock, if you're interested in giving that a go.  This code should have _some_ testing in the normal build.